### PR TITLE
Initial Test Plan for Shopping Cart

### PR DESCRIPTION
**Pull request by dustinhansen**

### CODE REVIEW

File: shopping_cart/test_plan.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Feature: Shopping Cart
+
+## Scenario: User adds item to cart
+  Given I am an User
+  When I go to an Item page
+  And I click "Add Item To Cart"
+  Then the quantity of items in my cart should increase by 1
+  And the warehouse inventory for that item should decrease by 1
```
Comment (djkotowski) [inline shopping_cart/test_plan.md]: As this is the front-end, the user won't be able to see the warehouse inventory.

File: shopping_cart/test_plan.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  And the cart information is saved to cookie
```
Comment (djkotowski) [inline shopping_cart/test_plan.md]: This is also less of a concern for the user.  Let the UI demonstrate if cart info is saved.

File: shopping_cart/test_plan.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  Given I am an User
+  And I have 1 or more items in my cart
+  And I am viewing the Cart page
+  When I enter a valid coupon code in the Coupon Code field
+  And click the "Apply Coupon Code" field
+  Then the corresponding amount is subtracted from my subtotal
+
+## Scenario: Revisiting a previous cart (authenticated users)
+  Given I am a Prior User
+  And I am logged in
```
Comment (djkotowski) [inline shopping_cart/test_plan.md]: What's the difference between L61-62 and `Given I am an User`?

File: shopping_cart/test_plan.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+# Feature: Shopping Cart
+
+## Scenario: User adds item to cart
+  Given I am an User
+  When I go to an Item page
```
Comment (djkotowski) [inline shopping_cart/test_plan.md]: You mix capitalization of "Item" throughout your feature.

File: shopping_cart/test_plan.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  Then the corresponding item is removed form my cart completely
+  And the warehouse inventory for that Item increases accordingly
+  And the cart information is saved to cookie
+  
+## Scenario: User changes product quantity in cart
+  Given I am an User
+  And I have 1 or more items in my cart
+  And I am viewing the Cart page
+  When I edit the Quantity field
+  And click the "Update Cart" button
```
Comment (djkotowski) [inline shopping_cart/test_plan.md]: L24-25 can be replaced with `When I change the quantity`.  No need to get into implementation specifics in your feature.  That's what the step definitions are for.

File: shopping_cart/test_plan.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  And I have 1 or more items in my cart
+  And I am viewing the Cart page
+  Then the subtotal displayed is a total of all Items in my cart (before tax)
+  And the total displayed is a total of all Items in my cart (after tax)
+
+## Scenario: Get estimated shipping costs for cart
+  Given I am an User
+  And I have 1 or more items in my cart
+  And I am viewing the Cart page
+  When I enter a ZIP code in the ZIP Code field
```
Comment (djkotowski) [inline shopping_cart/test_plan.md]: `in the ZIP Code field` is a bit superfluous here.

**Comment by djkotowski** [comment]
This is a good start.  Take a look at the comments and make changes where needed.  Some additional notes:
- You'll want to split these tests into separate features.  Adding quantity to the cart belongs in a different feature than shipping estimates.
- Instead of writing these in Markdown, write them as Cucumber features (*.feature).  Github will include syntax highlighting for them.
- Consider sad/bad paths when writing tests.

File: shopping_cart/test_plan.md (hunk 3 above)
Comment (dustinhansen) [inline shopping_cart/test_plan.md]: You could be a Prior User that built up a cart but never checked out. We would still have your prior cart data in cookie form.

**Pull request closed without merging**